### PR TITLE
Add `@here`, `@line`, and `@file`

### PR DESCRIPTION
**Pull request by V-FEXrt**

### CODE REVIEW

**Review by terpstra — APPROVED** [review]
That's how it's done, yep!

**Review by JakeSiFive — APPROVED** [review]
looks good!

File: src/parser/syntax.cpp [hunk 1]
```diff
@@ -361,8 +361,14 @@ const char *symbolExample(int symbol) {
       return "3.1415";
     case TOKEN_INTEGER:
       return "42";
-    case TOKEN_KW_HERE:
-      return "here";
+    case TOKEN_KW_MACRO_HERE:
+      return "@here";
```
[the remark this answers is not in the document]
Comment (V-FEXrt) [reply src/parser/syntax.cpp:365]: `TOKEN_KW_HERE` was just renamed to `TOKEN_KW_MACRO_HERE`

the lexemes `here` and `@here` both get tokenized to `TOKEN_KW_MACRO_HERE`

https://github.com/sifive/wake/blob/master/src/parser/lexer.re#L105-L106

**Comment by V-FEXrt** [comment]
> There's existing code that depends on the old here spelling, and I'm not sure we're able to write a general compatibility wrapper around @here, so we're going to need a migration strategy before we release the next version. It might be best to keep here for a while until everything is changed over.

`here` and `@here` will both be supported until a full migration is completed. Since the value is translated in the cst we can't do something nice like

`def here = @here` since the `@here` will resolve to the location of the compat file

**Pull request merged**